Services/Component/classes/class.ilObjComponentSettingsGUI.php

```diff
@@ -439,32 +432,24 @@ function showPlugin()
 				$form->addItem($non);
 			}
 		}
-	
+
 		$tpl->setContent($form->getHTML());
 	}
-	
-	/**
-	* Refresh plugins information
-	*/
-	function refreshPluginsInformation()
-	{
-		global $ilCtrl;
-		
-die ("ilObjComponentSettigsGUI::refreshPluginsInformation: deprecated");
 
+	/**
+	 * Install a plugin
+	 *
+	 * @return null
+	 */
+	protected function installPlugin() {
 		include_once("./Services/Component/classes/class.ilPlugin.php");
-		ilPlugin::refreshPluginXmlInformation();
-		
-		if ($_GET["mode"] == IL_COMP_SERVICE)
-		{
-			$ilCtrl->redirect($this, "listServices");
-		}
-		else
-		{
-			$ilCtrl->redirect($this, "listModules");
-		}
+		$pl = ilPlugin::getPluginObject($_GET["ctype"], $_GET["cname"],
+			$_GET["slot_id"], $_GET["pname"]);
```

Comment:
The frame above ist executeCommand. There, as always, is $this->$cmd() is used. If i move the usage one up i have to refactor this function too. So we decide to leave it as it is.